pkg/operator/k8sutil/configmap.go

```diff
@@ -36,3 +39,38 @@ func DeleteConfigMap(clientset kubernetes.Interface, cmName, namespace string, o
 	defaultWaitOptions := &WaitOptions{RetryCount: 20, RetryInterval: 2 * time.Second}
 	return DeleteResource(delete, verify, resource, opts, defaultWaitOptions)
 }
+
+// getConfigMap returns a configMap or error
+func getConfigMap(clientset kubernetes.Interface, cmName, namespace string) (*corev1.ConfigMap, error) {
+	cm := &corev1.ConfigMap{}
+	cm, err := clientset.CoreV1().ConfigMaps(namespace).Get(cmName, metav1.GetOptions{})
+	if err != nil {
+		return cm, err
+	}
+	return cm, nil
+}
+
+// GetOperatorSetting gets the operator setting from ConfigMap or Env Var
+// return defaultValue if setting is not found
+func GetOperatorSetting(clientset kubernetes.Interface, settingName, defaultValue string) (string, error) {
+	var settingValue string
+	// config must be in operator pod namespace
+	const rookCephOperatorSetting string = "rook-ceph-operator-config"
+	namespace := os.Getenv("POD_NAMESPACE")
+	cm, err := getConfigMap(clientset, rookCephOperatorSetting, namespace)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if settingValue = os.Getenv(settingName); settingValue != "" {
+				return settingValue, nil
+			}
+			return defaultValue, nil
+		}
+		return defaultValue, fmt.Errorf("error reading ConfigMap %q. %v", rookCephOperatorSetting, err)
+	}
+	if cm.Data != nil && cm.Data[settingName] != "" {
```

Comment:
The setting value might actually be empty. I think you want to check `val, ok := cm.Data[settingName]; ok`  and then use `val` if `ok`